### PR TITLE
refactor: make some type boundaries stricter.

### PR DESCRIPTION
**Pull request by shintaroNagata**
## Why

Even though Subtable Fields, Group Fields, and Form Layouts have a specific inner structure, these type definitions receive every `object` type as a parameter.

## What

1. Make type boundaries of parameters in `SubtableField`, `SubtableFieldProperty`, `SubtableLayout`, `GroupLayout`, and `RowLayout` stricter.
1. Add tests for these types.

## How to test

`yarn lint`

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/master/CONTRIBUTING.md)
- [ ] Updated documentation if it is required.
- [x] Added tests if it is required.
- [x] Passed `yarn lint` and `yarn test` on the root directory.

### CODE REVIEW

**Comment by shisama** [comment]
@shintaroNagata Looks good to me but I have a question.

**Comment by shisama** [comment]
Sorry, I made a mistake to close this.

File: packages/rest-api-client/src/KintoneFields/types/__test__/field.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import {
```
Comment (shisama) [inline packages/rest-api-client/src/KintoneFields/types/__test__/field.ts:1]: @shintaroNagata Tests you added don't work when I run `yarn test`. I purposely made type errors and run `yarn test` but it worked fine. When I run `yarn lint` or `yarn build`, it failed. Should `__test__` directory be named `__tests__`?

**Review by shisama — APPROVED** [review]
Thanks!

**Review by zaki-yama — APPROVED** [review]
👍

**Pull request merged**